DirectXHost.cs

```diff
@@ -8,230 +8,261 @@
+        [DllImport("user32.dll", SetLastError = true)]
+        static extern IntPtr FindWindowEx(IntPtr parentHandle, IntPtr childAfter, string className, string windowTitle);
+        private const int GWL_HWNDPARENT = -8;
+
+        private OverlayForm _overlayForm;
+        private PictureBox _overlayTarget;
+        private RenderForm _dxForm;
+        private GraphicsD3D11 _graphics;
+        private long time = DateTime.Now.Ticks;
+        private bool UserResized { get; set; }
+        private Size ClientSize { get; set; }
+
+        public bool IsDisposed { get; private set; }
+
+        #region Events
+        public event EventHandler HostWindowClosed;
+        public event EventHandler HostWindowCreated;
+        protected void OnHostWindowCreated()
+        {
+            if (HostWindowCreated != null)
+            {
+                HostWindowCreated(this._dxForm, new EventArgs());
+            }
+        }
+
+        protected void OnHostWindowClosed()
+        {
+            if (HostWindowClosed != null)
+            {
+                HostWindowClosed(this._dxForm, new EventArgs());
+            }
+        }
+        #endregion
+
+
+        public void Exit()
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task Run()
+        {
+            try
+            {
+                await Settings.Load();
+                InitializeRenderForm();
+
+                _graphics = new GraphicsD3D11();
+                _graphics.Initialise(_dxForm, true);
+                _dxForm.UserResized += (sender, args) =>
+                {
+                    var renderForm = sender as RenderForm;
+                    ClientSize = new Size(renderForm.ClientSize.Width, renderForm.ClientSize.Height);
+                    UserResized = true;
+                };
+                _overlayForm.Show();
+                RenderLoop.Run(_dxForm, RenderCallback, true);
+            }
+            catch (ArgumentException ex)
+            {
+                MessageBox.Show(string.Format("Exception running DirectX host window.\r\n{0}", ex.Message), "DirectX Host", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(string.Format("Exception running DirectX host window.\r\n{0}", ex.Message), "DirectX Host", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        // Set the bitmap object to the size of the screen
+        private Bitmap _bmpScreenshot;
+        Bitmap bmpScreenshot
+        {
+            get => _bmpScreenshot;
+            set
+            {
+                _bmpScreenshot = value;
+                _overlayTarget.Image = value;
+                updateImageBounds();
+            }
+        }
+
+        Graphics gfxScreenshot;
+        private void RenderCallback()
+        {
+            if (_dxForm.WindowState != FormWindowState.Minimized)

```

Comment:
What is the purpose of this logic inversion?

Reply:
WindowState can be Maximized (i.e. user snaps the window, keyboard shortcut, etc) which FormWindowState.Normal would ignore